blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -231,6 +233,22 @@ public Link getLink() {
         };
     }
 
+    @Override
+    public List<Object> getParameters() {
+        return getParameterDefinitions(job);
+    }
+
+    public static List<Object> getParameterDefinitions(Job job){
```

Comment:
@kzantow forgot where PMD work(API to get pipeline schema) has gone in, which plugin it is in? If introspection can give us parameter name default value and type for a given pipeline step, thats pretty much what we need to build it's schema. To me the bigger issue is associating which step in schema would correspond to which step in executed pipeline. There isn't an id that gets carried over from schema all the way to executed pipeline.